src/apps/logs/crud/notification.py

```diff
@@ -0,0 +1,101 @@
+import json
+
+from sqlalchemy import select
+from sqlalchemy.engine import Result
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm import Query
+
+from apps.logs.db.schemas import NotificationLogSchema
+from apps.logs.domain import (
+    NotificationLogCreate,
+    NotificationLogQuery,
+    PublicNotificationLog,
+)
+from apps.logs.errors import NotificationLogAlreadyExist
+from infrastructure.database.crud import BaseCRUD
+
+__all__ = ["NotificationLogCRUD"]
+
+
+class NotificationLogCRUD(BaseCRUD[NotificationLogSchema]):
+    schema_class = NotificationLogSchema
+
+    async def all(
+        self, query_set: NotificationLogQuery
+    ) -> list[PublicNotificationLog]:
+        """Return all notification logs where the user and device exists."""
+        query: Query = (
+            select(self.schema_class)
+            .where(
+                self.schema_class.user_id == query_set.user_id
+                and self.schema_class.device_id == query_set.device_id
+            )
+            .order_by(self.schema_class.created_at.desc())
+        )
+        if query_set.limit:
+            query = query.limit(query_set.limit)
+
+        result: Result = await self._execute(query)
+        results: list[PublicNotificationLog] = result.scalars().all()
+
+        return [PublicNotificationLog.from_orm(log) for log in results]
+
+    async def save(
+        self, schema: NotificationLogCreate
+    ) -> PublicNotificationLog:
+        """Return notification log instance and the created information."""
+
+        notif_desc_upd = True
+        notif_in_queue_upd = True
+        sched_notif_upd = True
```

Comment:
I would recommend using full namings instead of short ones. But it is definitely up to you

Reply:
Because names are quite long, it didn't fit into max-line-length rule of the linter.